### ISSUE TITLE
Availablity of Google Cloud Storage Resolver like AwsS3Resolver

### ISSUE THREAD

**Issue by mappedinn** [liip/LiipImagineBundle#666]
Hi,

LiipImagineBundle does not provide a Resolver for Google Cloud Storage like for AWS S3. This resolver is needed for enabling the application of LiipImagineBundle filters on the images I stored on Google Cloud Storage. This resolver will be used in this a way similar to aws s3:

```
liip_imagine.cache.resolver.google_cloud_storage:
    class: Liip\ImagineBundle\Imagine\Cache\Resolver\GoogleCloudStorageResolver
    arguments:
        - "@app.google_cloud_storage.service"
        - %gcs_bucket%
    tags:
        - {name: 'liip_imagine.cache.resolver', resolver: 'resolver.google_cloud_storage'}
```

To do so, I do need to create that class `GoogleCloudStorageResolver` as described in [this link on cache resolvers](http://symfony.com/doc/current/bundles/LiipImagineBundle/cache-resolvers.html). I know that I have define the methods of the `ResolverInterface` but I don't know from where to start. 

```
public function isStored($path, $filter);
public function resolve($path, $filter);
public function store(BinaryInterface $binary, $path, $filter);
public function remove(array $paths, array $filters);
```

If there is any link, I will be happy.

Thanks

**Comment by mappedinn** [comment]
Merry christmas @lsmith77 @makasim 
**One question:** is LiipImagineBundle planning to develop  Google Cloud Storage Resolver like AwsS3Resolver?

Best,

**Comment by makasim** [comment]
No, I dont have such plans cuz I dont work with google. If you work with Google Cloud Storage, maybe you can contribute it?

**Comment by mappedinn** [comment]
@makasim thank you very much. I will try my best.

**Comment by wieger** [comment]
@mappedinn Hi, I'm trying to achieve the same thing, get liip_imagine on top of Google Cloud Storage. Did you make any progress on this and would you be willing to open-source your solution?

**Comment by mappedinn** [comment]
@wieger Hi, my solution is described in details here https://github.com/liip/LiipImagineBundle/issues/698

I did not managed to make it work correctly, but it seems that @girayk made it work!

Keep me updated...

**Comment by michellesanver** [comment]
I do not believe this is something for this bundle to solve, also: Closed in favour of the discussion in #698
